File: crates/rootlight-bench/tests/fixtures/semantic-holdout/v1/sources/go/main/main.go
package main

import (
    "example/holdout/dep_a"
    "example/holdout/dep_b"
)

type FixtureBase interface{}
type FixtureDerived interface { FixtureBase }

type FixtureProtocol interface {
    FixtureBase
    FixtureNode09()
    FixtureNode10()
}

var _ FixtureProtocol = dep_a.FixtureVariantA{}
var _ FixtureProtocol = dep_b.FixtureVariantB{}

func ExecuteFixture(receiver FixtureProtocol) {
    dep_a.FixtureNode01()
    dep_a.FixtureNode02()
    dep_a.FixtureNode03()
    dep_a.FixtureNode04()
    dep_a.FixtureNode05()
    dep_a.FixtureNode06()
    dep_a.FixtureNode07()
    dep_a.FixtureNode08()
    receiver.FixtureNode09()
    receiver.FixtureNode10()
    FixtureNode11()
    FixtureNode12()
}

File: crates/rootlight-bench/tests/fixtures/semantic-holdout/v1/sources/javascript/main.js
import { FixtureVariantA, fixtureNode01 } from "./dep-a";
import { fixtureNode02 } from "./dep-a";
import { fixtureNode03 } from "./dep-a";
import { fixtureNode04 } from "./dep-a";
import { fixtureNode05 } from "./dep-a";
import { fixtureNode06 } from "./dep-a";
import { fixtureNode07 } from "./dep-a";
import { fixtureNode08 as fixtureNode08Local } from "./dep-a";
import { FixtureVariantB } from "./dep-b";

/** @param {FixtureVariantA | FixtureVariantB} receiver */
export function executeFixture(receiver) {
    fixtureNode01();
    fixtureNode02();
    fixtureNode03();
    fixtureNode04();
    fixtureNode05();
    fixtureNode06();
    fixtureNode07();
    fixtureNode08Local();
    receiver.fixtureNode09();
    receiver.fixtureNode10();
    fixtureNode11();
    fixtureNode12();
}
